tgui/packages/tgui/interfaces/PaiInterface.tsx

```diff
@@ -0,0 +1,540 @@
+  const { data } = useBackend<PaiInterfaceData>(context);
+  const { directives, master } = data;
+  return (
+    <Stack fill vertical>
+      <Stack.Item>
+        <Section title="Logic Core">
+          <Box color="label">
+            {DIRECTIVE_COMPREHENSION}
+            <br />
+            <br />
+            {DIRECTIVE_ORDER}
+          </Box>
+        </Section>
+      </Stack.Item>
+      <Stack.Item grow>
+        <Section fill scrollable title="Directives">
+          {!master.name ? (
+            'None.'
+          ) : (
+            <LabeledList>
+              <LabeledList.Item label="Prime">
+                Serve your master.
+              </LabeledList.Item>
+              <LabeledList.Item label="Supplemental">
+                <Box wrap>{directives}</Box>
+              </LabeledList.Item>
+            </LabeledList>
+          )}
+        </Section>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const InstalledDisplay = (_, context) => {
+  const [softwareSelected, setSoftwareSelected] = useSharedState(
+    context,
+    'software',
+    ''
+  );
+  const setSoftwareHandler = (software: string) => {
+    setSoftwareSelected(software);
+  };
+
+  return (
+    <Stack fill vertical>
+      <Stack.Item grow>
+        <InstalledSoftware softwareHandler={setSoftwareHandler} />
+      </Stack.Item>
+      <Stack.Item grow={2}>
+        <InstalledInfo software={softwareSelected} />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const InstalledSoftware = (props, context) => {
+  const { data } = useBackend<PaiInterfaceData>(context);
+  const { installed } = data.software;
+  const { softwareHandler } = props;
+
+  return (
+    <Section fill scrollable title="Installed Software">
+      {!installed.length ? (
+        <NoticeBox>Nothing installed!</NoticeBox>
+      ) : (
+        installed.map((software, index) => {
+          return (
+            <Button key={software} onClick={() => softwareHandler(software)}>
+              {software.replace(/(^\w{1})|(\s+\w{1})/g, (letter) =>
+                letter.toUpperCase()
+              )}
+            </Button>
+          );
+        })
+      )}
+    </Section>
+  );
+};
+
+const InstalledInfo = (props, context) => {
+  const { act, data } = useBackend<PaiInterfaceData>(context);
+  const { software } = props;
+
+  return (
+    <Section
+      fill
+      scrollable
+      title={
+        !software
+          ? 'Select a Program'
+          : software.replace(/(^\w{1})|(\s+\w{1})/g, (letter) =>
+              letter.toUpperCase()
+            )
+      }>
+      {software && (
+        <Stack fill vertical>
+          <Stack.Item>
+            <SoftwareInfo software={software} />
+          </Stack.Item>
+          <Stack.Item>
+            <SoftwareButtons software={software} />
+          </Stack.Item>
+        </Stack>
+      )}
+    </Section>
+  );
+};
+
+const SoftwareInfo = (props) => {
+  const { software } = props;
+
+  switch (software) {
+    case 'crew manifest':
```

Comment:
You should make this a const object, rather than a huge switch-case.

Reply:
yandere dev moment